library/src/app/components/trade/trade.component.ts

```diff
@@ -1,10 +1,256 @@
+import { Constants } from '../../../../../src/shared/constants/constants';
+import { AssetPipe } from '../../../../../src/shared/pipes/asset.pipe';
+import { compareObjects } from '../../../../../src/shared/utility/compare-object';
+import { symbolSplit } from '../../../../../src/shared/utility/symbol-split';
+import { symbolBuild } from '../../../../../src/shared/utility/symbol-build';
+import { TradeBankModel } from '@cybrid/cybrid-api-bank-angular/model/trade';
+import SideEnum = PostQuoteBankModel.SideEnum;
+
+interface Quote {
+  asset: AssetBankModel;
+  counterAsset: AssetBankModel;
+  amount: number;
+  value: number;
+}
 
 @Component({
   selector: 'app-trade',
   templateUrl: './trade.component.html',
-  styleUrls: ['./trade.component.scss']
+  styleUrls: ['./trade.component.scss'],
+  encapsulation: ViewEncapsulation.ShadowDom,
+  changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class TradeComponent {
-  constructor() {}
+export class TradeComponent implements OnInit, OnDestroy {
+  compareObjects = compareObjects;
+
+  assetList: AssetBankModel[] = [];
+  filteredAssetList: AssetBankModel[] = [];
+
+  price: SymbolPriceBankModel = {};
+
+  quoteGroup: FormGroup = new FormGroup({
+    asset: new FormControl(),
+    amount: new FormControl()
+  });
+
+  quote: Quote = {
+    asset: Constants.ASSET,
+    counterAsset: Constants.COUNTER_ASSET,
+    amount: 0,
+    value: 0
+  };
+
+  side: TradeBankModel.SideEnum = SideEnum.Buy;
+  input: AssetBankModel.TypeEnum = 'fiat';
+
+  quote$ = new BehaviorSubject<Quote>(this.quote);
+
+  isLoading$ = new BehaviorSubject(true);
+  isRecoverable$ = new BehaviorSubject(true);
+  unsubscribe$ = new Subject();
+
+  constructor(
+    private errorService: ErrorService,
+    private eventService: EventService,
+    private assetService: AssetService,
+    public configService: ConfigService,
+    private pricesService: PricesService,
+    private assetPipe: AssetPipe,
+    private route: ActivatedRoute
+  ) {}
+
+  ngOnInit() {
+    this.eventService.handleEvent(
+      LEVEL.INFO,
+      CODE.COMPONENT_INIT,
+      'Initializing trade component'
+    );
+    this.getRouterParams();
+    this.getAssets();
+    this.initQuoteGroup();
+    this.getPrice();
+    this.refreshData();
+  }
+
+  ngOnDestroy() {
+    this.unsubscribe$.next('');
+    this.unsubscribe$.complete();
+  }
+
+  getRouterParams(): void {
+    this.route.queryParams
+      .pipe(
+        takeUntil(this.unsubscribe$),
+        map((params) => {
+          this.quote.asset = JSON.parse(params['asset']);
+          this.quote.counterAsset.code = symbolSplit(params['symbol_pair'])[1];
+        })
+      )
+      .subscribe();
+  }
+
+  getAssets(): void {
+    this.assetService
+      .getAssets$()
+      .pipe(
+        take(1),
+        map((assets) => {
+          this.assetList = assets;
+          this.filteredAssetList = assets.filter((asset) => {
+            return asset.type == 'crypto';
```

Comment:
Suggest you read this from the asset list response and not set it based on asset/ counter-asset

Reply:
I agree, it's a temp solution. When I add the customerGUID @input and we can query for an account list i'll refactor this and put all available accounts in the currency select. We need the data to determine whether to add USD or CAD or both.